src/transformers/models/bark/processing_bark.py

```diff
@@ -0,0 +1,295 @@
+# coding=utf-8
+# Copyright 2023 The Suno AI Authors and The HuggingFace Inc. team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Processor class for Bark
+"""
+import os
+import warnings
+from typing import Optional, Union
+
+import numpy as np
+
+from ...processing_utils import ProcessorMixin
+from ...utils import (
+    TensorType,
+    is_flax_available,
+    is_jax_tensor,
+    is_numpy_array,
+    is_tf_available,
+    is_torch_available,
+)
+from ...utils.hub import get_file_from_repo
+from ..auto import AutoTokenizer
+
+
+class BarkProcessor(ProcessorMixin):
+    r"""
+    Constructs a Bark processor which wraps a text tokenizer and optional Bark voice presets into a single processor.
+
+    Args:
+        tokenizer ([`PreTrainedTokenizer`]):
+            An instance of [`PreTrainedTokenizer`].
+        speaker_embeddings_dict (`Dict[np.ndarray]`, *optional*, defaults to `None`):
+            Optional speaker embeddings dictionary. The keys follow the following pattern:
+            `"{voice_preset_name}_{prompt_key}"`. For example: `"en_speaker_1_semantic_prompt"` or
```

Comment:
Would a nested dictionary be cleaner here?
```python
{"en_speaker_1": {"semantic_prompt": ...}, "en_speaker_2": {"semantic_prompt": ...}, ...}
```

Reply:
The only thing is that you can't save a `.npz` file as a nested dictionnary so it would add a complicated layer to the processor every time that an user loads `from_pretrained` or `save_pretrained`.

Reply:
Does it not? See for example:
```python
import numpy as np
from tempfile import TemporaryFile

dictionary = {"en_speaker_1": {"semantic_prompt": np.ones(10)}}

outfile = TemporaryFile()
np.savez(outfile, dictionary)

_ = outfile.seek(0) # Only needed here to simulate closing & reopening file
npz_file = np.load(outfile, allow_pickle=True)

print("Files: ", npz_file.files)
print("Key `arr_0`: ", npz_file["arr_0"])
```

Reply:
At the end of the day, I've done it like that, but am worry about the security of `pickle`